### PR TITLE
Cubic CO2 sensors support (CM110x)

### PR DESCRIPTION
**Pull request by mrekin**
## Description:

Adding support for CM110x CO2 sensors by CUBIC (serial interface).
Code base got from Mhz19 sensor and adopted for CM110x.

Need to define USE_CM110x for usage.

Protocol description:
https://en.gassensor.com.cn/CO2Sensor/list.html

Tested with CM1107 sensor, expected support CM1106 and CM1109 (Cubic  has a wide list of CO2 sensors, but most of them has similar  serial protocol in documentation).
Most sensors returns only PPM, no temp or humidity. Only one model I found (CM1107T) which return temp (code implemented but not tested).

Tested on esp8266, for ESP32 only builds tested (tasmota32 and tasmota32-bluetooth).

!!! Code **re-use `Settings->SensorBits1.mhz19b_abc_disable`** setting for ABC controlling.
If this bad decision - I can add new one.

Implemented commands:
1. ABC enabling/disabling
2. ABC enabling with custom ABC period set.
3. Manual calibration (400ppm level)
4. Reading SW version and serial number.
5. Set ppm limits (in driver, not in device - different models has different ppm ranges) and filter mode.

Code brings additional 2k7 Flash usage.
Without USE_CM110x
```
RAM:   [=====     ]  50.0% (used 40940 bytes from 81920 bytes)
Flash: [======    ]  61.6% (used 630720 bytes from 1023984 bytes)
Building .pio\build\tasmota\firmware.bin
```
With USE_CM110x
```
RAM:   [=====     ]  50.4% (used 41292 bytes from 81920 bytes)
Flash: [======    ]  61.9% (used 633476 bytes from 1023984 bytes)
Building .pio\build\tasmota\firmware.bin
```
Docs will be prepeared little bit later.

**Related issue (if applicable):** fixes #<Tasmota issue number goes here>

## Checklist:
  - [x] The pull request is done against the latest development branch
  - [x] Only relevant files were touched
  - [x] Only one feature/fix was added per PR and the code change compiles without warnings
  - [x] The code change is tested and works with Tasmota core ESP8266 V.2.7.4.9
  - [ ] The code change is tested and works with Tasmota core ESP32 V.2.0.2.1
  - [x] I accept the [CLA](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md#contributor-license-agreement-cla).

_NOTE: The code change must pass CI tests. **Your PR cannot be merged unless tests pass**_

Some pics:
WebUI
image

HA card
image

Sensor (with board)
image

### CODE REVIEW

File: tasmota/language/en_GB.h [hunk 1]
```diff
@@ -852,6 +852,10 @@
 #define D_GPIO_SHIFT595_RCLK   "74x595 RCLK"
 #define D_GPIO_SHIFT595_OE     "74x595 OE"
 #define D_GPIO_SHIFT595_SER    "74x595 SER"
+#define D_SENSOR_CM11_TX       "CM110x TX"
```
Comment (Jason2866) [inline tasmota/language/en_GB.h:855]: Remove duplicate

Comment (mrekin) [reply tasmota/language/en_GB.h:855]: Thanks. Done

File: tasmota/tasmota_template_legacy.h [hunk 2]
```diff
@@ -87,6 +87,8 @@ enum LegacyUserSelectablePins {
   GPI8_LED4_INV,
   GPI8_MHZ_TXD,        // MH-Z19 Serial interface
   GPI8_MHZ_RXD,        // MH-Z19 Serial interface
+  GPI8_CM11_TXD,       // CM110x Serial interface
+  GPI8_CM11_RXD,       // CM110x Serial interface
```
Comment (arendst) [inline tasmota/tasmota_template_legacy.h:91]: Changes in this file are not supported. Pls remove any changes. See top of this file.

Comment (mrekin) [reply tasmota/tasmota_template_legacy.h]: Done.

**Pull request merged**